pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -60,6 +60,10 @@ func validateInstanceSpec(spec *sc.InstanceSpec, fldPath *field.Path, create boo
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("planName"), spec.PlanName, msg))
 	}
 
+	if spec.Parameters != nil && spec.AlphaSecretParameters != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("alphaSecretParameters"), spec.AlphaSecretParameters, "only one of 'Parameters' and 'AlphaSecretParameters' can be set at a time"))
```

Comment:
@duglin @arschles re: merging
I am not against support of both secret and non-secret parameters simultaneously in general.
But I would prefer to keep this out of scope of this PR. This PR is about design for secret parameters support, and lack of merging strategy makes it easier to review.
A separate PR with adding merging support is fine with me, and will also be focused on single feature easy to review, and won't require any breaking API changes.